### PR TITLE
Avoid random guest names and make guest state more visible

### PR DESCRIPTION
**Pull request by juliushaertl**
* Resolves: https://github.com/nextcloud/text/issues/652
* Target version: master 

### Summary

- Make sure that guest users use "Guest" as a default name instead of random vegetable names
- Allow guests to reset their name to the default after changing it initially
- Adjust the avatar style for guests so that their avatar just uses a grey background with the first character of the name
  - Refactor the avatar handling in the session menu a bit to have a shared AvatarWrapper that takes care of the user/guest rendering and can be reused

This is the same behaviour as it is already in place in the Talk app.

First 3 commits can be backported separately to address the confusing default names
| Before | After |
| --- | --- |
| Screenshot 2022-05-25 at 14 38 30 | Screenshot 2022-05-25 at 14 38 02 |

### CODE REVIEW

**Comment by juliushaertl** [comment]
Rebased and added screenshots for comparison

File: src/components/AvatarWrapper.vue [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		sessionColorBorder() {
+			return {
+				...this.sessionBackground,
+				'border-color': this.session.color,
+				'border-width': '2px',
+				'border-style': 'solid',
+				'--size': this.size + 'px',
+				'--font-size': this.size / 2 + 'px',
+			}
+		},
```
Comment (vinicius73) [inline src/components/AvatarWrapper.vue:43]: This is not only for border.

Comment (juliushaertl) [reply src/components/AvatarWrapper.vue]: Yes, renamed :)

**Review by luka-nextcloud — APPROVED** [review]

**Comment by juliushaertl** [comment]
Test succeeded after restart. Might still be worth to monitor if that is flaky and needs some improvements:

https://dashboard.cypress.io/projects/hx9gqy/runs/4000/test-results/9c8707e6-4ab8-4124-a226-e360db96653c

**Comment by juliushaertl** [comment]
Let's skip this backport since this is more an enhancement and would bring a large changeset to the stable branches.

**Pull request merged**